test/test_active.py

```diff
@@ -0,0 +1,279 @@
+            if func in ["mul", "mul_"]:
+                continue
+            for tensor_type in [lambda x: x, MPCTensor]:
+                tensor1 = self._get_random_test_tensor(is_float=True)
+                tensor2 = self._get_random_test_tensor(is_float=True)
+                encrypted = MPCTensor(tensor1)
+                encrypted2 = tensor_type(tensor2)
+
+                reference = getattr(tensor1, func)(tensor2)
+                encrypted_out = getattr(encrypted, func)(encrypted2)
+
+                self._check(
+                    encrypted_out,
+                    reference,
+                    "%s %s failed"
+                    % ("private" if tensor_type == MPCTensor else "public", func),
+                )
+                if "_" in func:
+                    # Check in-place op worked
+                    self._check(
+                        encrypted,
+                        reference,
+                        "%s %s did not modify input as expected"
+                        % ("private" if tensor_type == MPCTensor else "public", func),
+                    )
+                else:
+                    # Check original is not modified
+                    self._check(
+                        encrypted,
+                        tensor1,
+                        "%s %s modified input"
+                        % ("private" if tensor_type == MPCTensor else "public", func),
+                    )
+
+                # Check encrypted vector with encrypted scalar works.
+                tensor1 = self._get_random_test_tensor(is_float=True)
+                tensor2 = self._get_random_test_tensor(is_float=True, size=(1,))
+                encrypted1 = MPCTensor(tensor1)
+                encrypted2 = MPCTensor(tensor2)
+                reference = getattr(tensor1, func)(tensor2)
+                encrypted_out = getattr(encrypted1, func)(encrypted2)
+                self._check(encrypted_out, reference, "private %s failed" % func)
+
+        # test square
+        # tensor = self._get_random_test_tensor(is_float=True)
+        # reference = tensor * tensor
+        # encrypted = MPCTensor(tensor)
+        # encrypted_out = encrypted.square()
+        # self._check(encrypted_out, reference, "square failed")
+
+        # # Test radd, rsub, and rmul
+        # reference = 2 + tensor1
+        # encrypted = MPCTensor(tensor1)
+        # encrypted_out = 2 + encrypted
+        # self._check(encrypted_out, reference, "right add failed")
+
+        # reference = 2 - tensor1
+        # encrypted_out = 2 - encrypted
+        # self._check(encrypted_out, reference, "right sub failed")
+
+        # reference = 2 * tensor1
+        # encrypted_out = 2 * encrypted
+        # self._check(encrypted_out, reference, "right mul failed")
+
+    def test_div(self):
+        """Tests division of encrypted tensor by scalar and tensor."""
+        for function in ["div", "div_"]:
+            raise NotImplementedError
```

Comment:
From CircleCI:

This test is failing here since we are raising `NotImplementedError`. I don't think we should expect this test to work based on the current code. I would just take this test out (lines 195-212).